doc/design/eventdriven-callbacks.md

```diff
@@ -0,0 +1,247 @@
+## Problem Description
+
+We require a public API so that a developer can register callbacks for certain
+events like the arrival of a sample.
+
+## Name of the design element
+
+ - ReactAl = React And Listen
+ - Reactor
+ - Eventler
+ - LiAR - Listen And React
+
+
+## Terminology
+
+ - **event driven** a one time reaction which is caused directly by an event.
+      Example: a new sample has been delivered to a subscriber.
+ - **state driven** a repeating reaction which is caused by a specific structural 
+      state until the state changes.
+      Example: a subscriber has stored samples which were not inspected by the user.
+ - **robust API** we use this term to describe an API which is hard 
```

Comment:
The idea behind this definition was to state that certain features are not introduced because they are nice but because they can make the daily developer live hard since they had to pay attention to a lot of restrictions. Sometimes they are maybe even not aware that the code is running inside of such a Reactor.

@MatthiasKillat 
Robust is the word which fits in this particular case best. But I am open for other suggestions! I know it is vaguely defined but again, I am again open for suggestions!